crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2504,13 +2587,15 @@ pub enum KnownFunction {
     ConstraintFunction(KnownConstraintFunction),
     /// `builtins.reveal_type`, `typing.reveal_type` or `typing_extensions.reveal_type`
     RevealType,
+    /// `builtins.len`
+    Len,
 }
 
 impl KnownFunction {
     pub fn constraint_function(self) -> Option<KnownConstraintFunction> {
         match self {
             Self::ConstraintFunction(f) => Some(f),
-            Self::RevealType => None,
+            _ => None,
```

Comment:
I would prefer that we enumerate these all explicitly. It means that the compiler forces us to think about whether the new function we're adding support for is a constraint function or not whenever we add a new `KnownFunction`; this is very useful!

```suggestion
            Self::RevealType | Self::Len => None,
```